frontend/test/__support__/settings.ts

```diff
@@ -0,0 +1,13 @@
+import MetabaseSettings from "metabase/lib/settings";
+import { createMockSettings } from "metabase-types/api/mocks";
+import { createMockSettingsState } from "metabase-types/store/mocks";
+import type { Settings } from "metabase-types/api";
+
+export function mockSettings(params: Partial<Settings> = {}) {
+  const settings = createMockSettings(params);
+  const state = createMockSettingsState({ values: settings });
+
+  MetabaseSettings.setAll(settings);
+
+  return { state };
```

Comment:
Do you plan to return something else other than `state` in the future?

other `createXXXXMock` return the object directly without having to destructure or get the mock value from the property. Would it be easier to just return `state` here?